utils/s2n_mem.c

```diff
@@ -51,7 +52,7 @@ static int s2n_mem_init_impl(void)
     POSIX_ENSURE_GT(sysconf_rc, 0);
 
     /* page_size must be a valid uint32 */
-    POSIX_ENSURE_LTE(sysconf_rc, UINT32_MAX);
+    POSIX_ENSURE(sysconf_rc <= ((LONG_MAX > UINT32_MAX) ? (long) UINT32_MAX : LONG_MAX), S2N_ERR_SAFETY);
 
```

Comment:
Nit: This is a pretty intense check. It'd probably be cleaner if you split it up:
```suggestion
    long max_page_size = MIN(UINT32_MAX, LONG_MAX);
    POSIX_ENSURE_LTE(sysconf_rc <= max_page_size));

```